onedal/datatypes/_data_conversion.py

```diff
@@ -43,32 +46,37 @@ def to_table(*args):
     from ..common._policy import _HostInteropPolicy
 
     def _convert_to_supported_impl(policy, *data):
+        def func(x):
+            return x
+
         # CPUs support FP64 by default
         if isinstance(policy, _HostInteropPolicy):
-            return data
+            return _apply_and_pass(func, *data)
 
         # It can be either SPMD or DPCPP policy
         device = policy._queue.sycl_device
 
         def convert_or_pass(x):
-            if x.dtype is np.float64:
+            if x.dtype == np.float64:
+                warnings.warn("Data will be converted into float32 from "
+                              "float64 because device does not support it",
+                              RuntimeWarning, )
                 return x.astype(np.float32)
             else:
                 return x
 
         if not device.has_aspect_fp64:
-            return (convert_or_pass(x) for x in data)
-        else:
-            return data
+            func = convert_or_pass
+
+        return _apply_and_pass(func, *data)
+
 else:
     def _convert_to_supported_impl(policy, *data):
-        return data
+        def func(x):
+            return x
 
+        return _apply_and_pass(func, *data)
 
-def _convert_to_supported(policy, *data):
-    res = _convert_to_supported_impl(policy, *data)
 
-    if len(data) == 1:
-        return res[0]
-    else:
-        return res
+def _convert_to_supported(policy, *data):
+    return _convert_to_supported_impl(policy, *data)
```

Comment:
Meaningless, rename _convert_to_supported_impl to _convert_to_supported and remove this

Reply:
Done